src/fitnesse/junit/JUnitRunNotifierResultsListener.java

```diff
@@ -51,15 +51,22 @@ public void testComplete(TestPage test, TestSummary testSummary) {
     increaseCompletedTests();
     if (firstFailure != null) {
       notifier.fireTestFailure(new Failure(descriptionFor(test), firstFailure));
+      fireTestFinishedFor(test);
     } else if (testSummary.getExceptions() > 0) {
       notifier.fireTestFailure(new Failure(descriptionFor(test), new Exception("Exception occurred on page " + test.getFullPath())));
+      fireTestFinishedFor(test);
     } else if (testSummary.getWrong() > 0) {
       notifier.fireTestFailure(new Failure(descriptionFor(test), new AssertionError("Test failures occurred on page " + test.getFullPath())));
+      fireTestFinishedFor(test);
     } else {
-      notifier.fireTestFinished(descriptionFor(test));
+      fireTestFinishedFor(test);
     }
```

Comment:
We could do one single fireTestFinishedFor() after the if statement I suppose. The docs say a fireTestFinished event should be executed for every fireTestStarted event.